### PR TITLE
[Performance, Hardware] MoE weights padding to AMD MI300x GPUs

### PR DESCRIPTION
**Pull request by HaiShaw**
## Motivation

Padding MoE weights (last dim) to minimize Memory Channel Contention (only to AMD Instinct GPUs)
Test shows approximate performance boost of prefill +2.2%, decode +3.0% for Grok-1 on setting: b32/i1024/o512

## Modifications

As mentioned: fused_moe.py and layer.py
To enable this feature, set binary flag `MOE_PADDING=1` at command line, or `export MOE_PADDING=1` in console.

## Checklist

- [+] Format your code according to the [Contributor Guide](https://github.com/sgl-project/sglang/blob/main/docs/contributor_guide.md).
- [+] Add unit tests as outlined in the [Contributor Guide](https://github.com/sgl-project/sglang/blob/main/docs/contributor_guide.md).
- [+] Update documentation as needed, including docstrings or example tutorials.

### CODE REVIEW

File: python/sglang/srt/layers/fused_moe/__init__.py [hunk 1]
```diff
@@ -1 +1,2 @@
 from sglang.srt.layers.fused_moe.layer import FusedMoE, FusedMoEMethodBase
+from fused_moe import padding_size
```
Comment (merrymercy) [inline python/sglang/srt/layers/fused_moe/__init__.py]: use absolute path

File: python/sglang/srt/layers/fused_moe/layer.py [hunk 2]
```diff
@@ -1,9 +1,12 @@
 # Adapted from
 # https://github.com/vllm-project/vllm/tree/v0.5.4/vllm/model_executor/layers/fused_moe
+import os
 from abc import abstractmethod
 from typing import List, Optional, Tuple
 
 import torch
+import torch.nn.functional as F
+from fused_moe import padding_size
```
Comment (merrymercy) [inline python/sglang/srt/layers/fused_moe/layer.py]: Use absolute path and sort the imports correctly

**Review by merrymercy — CHANGES_REQUESTED** [review]

**Comment by merrymercy** [comment]
Please fix the CI.

**Comment by HaiShaw** [comment]
@merrymercy Fixed the CI just now. Thanks!

File: python/sglang/srt/layers/fused_moe/layer.py [hunk 3]
```diff
@@ -572,6 +588,18 @@ def process_weights_after_loading(self, layer: Module) -> None:
                     start += shard_size
 
             layer.w13_scale = torch.nn.Parameter(max_w13_scales, requires_grad=False)
+            # If ROCm, apply weight padding (min. Mem channel contention) only if set
+            if is_hip() and bool(int(os.getenv("MOE_PADDING", "0"))):
```
Comment (merrymercy) [inline python/sglang/srt/layers/fused_moe/layer.py:592]: move all `is_hip` under a single branch.
e.g., L555

**Review by merrymercy — CHANGES_REQUESTED** [review]

File: python/sglang/srt/layers/fused_moe/layer.py (hunk 3 above)
Comment (HaiShaw) [reply python/sglang/srt/layers/fused_moe/layer.py:592]: @merrymercy understand, the order of data crunching makes me intend to keep dummy padding at very last to avoid error prone situation from intervening `normalize_`, `_dequantize`. `_fp8_quant`, etc., and easier to read.

**Review by merrymercy — APPROVED** [review]
